### DISCUSSION TITLE
bin/vite dev error in Rails 5.1 (<< wrong number of arguments)

### DISCUSSION THREAD

**Discussion by GabrielMahan** [ElMassimo/vite_ruby#69] [Q&A] [answered]
Hope y'all will excuse my ignorance of the project, but I thought I'd try out vite on an existing rails project, and I followed the steps, but when I got to ` bin/vite dev` I was met with an argument error .

```
gems/2.5.0/gems/vite_ruby-1.2.6/lib/vite_ruby/runner.rb:34:in `<<': wrong number of arguments (given 0, expected 1) (ArgumentError)
```

The other bin commands like `bin/vite version` work fine

```
bin/vite present?: true
vite_ruby: 1.2.6
vite_rails: 2.0.10
rails: 5.1.6.2
node: v16.0.0
npm: 7.10.0
yarn: 1.22.10
pnpm:
ruby: ruby 2.5.8p224 (2020-03-31 revision 67882) [x86_64-darwin19]

installed packages:
[project]@ /Users/gabrielmahan/[project]
├─┬ vite-plugin-ruby@2.0.3
│ └── vite@2.2.3 deduped
└── vite@2.2.3
```

**Comment by ElMassimo** [comment]
Might be related with Ruby 2.5—from the error it seems like `append` can't take an empty argument.

I'm running tests in 2.5 but perhaps the command is always being passed some arguments. I'll take a look tomorrow, the fix should be straightforward.

**Reply by ElMassimo** [reply]
@GabrielMahan Tested on `ruby 2.5.8p224 (2020-03-31 revision 67882) [x86_64-darwin19]` (same version).

The error is caused by how `activesupport` monkeypatches `Array`'s `append` method–in [`rails 5.1.6.2`](https://github.com/rails/rails/blob/5-1-stable/activesupport/lib/active_support/core_ext/array/prepend_and_append.rb#L2-L3) the `<<` method is aliased as `append` but this is not consistent with Ruby where it's an alias for `push`.

As a result, it can't handle a variable number of arguments, only one.

This bug was fixed in [`rails 5.2`](https://github.com/rails/rails/blob/5-2-stable/activesupport/lib/active_support/core_ext/array/prepend_and_append.rb#L4-L5), and in higher versions the method is no longer monkeypatched.

I will release a version that uses `push` instead of `append` in order to provide compatibility for `5.1`.

**Comment by ElMassimo** [comment, marked as answer]
The error is caused by how `activesupport` monkeypatches `Array`'s `append` method–in [`rails 5.1.6.2`](https://github.com/rails/rails/blob/5-1-stable/activesupport/lib/active_support/core_ext/array/prepend_and_append.rb#L2-L3) the `<<` method is aliased as `append` overriding Ruby stdlib method which is an alias for `push`.

As a result, it can only receive a single argument, instead of a variable number of arguments.

<hr/>

[Released `vite_ruby, '1.2.7'`](https://github.com/ElMassimo/vite_ruby/blob/main/vite_ruby/CHANGELOG.md#127-2021-04-28) which uses `push` instead of `append` to avoid this incompatibility in `rails 5.1.6.2`.

You can upgrade by running `bundle update vite_ruby`.

**Reply by GabrielMahan** [reply]
Dang @ElMassimo you're fast! Thank you and great work!
